src/api.js

```diff
@@ -23,14 +23,16 @@ const API = {
       'onRequestCompleted',
       'requestData',
       'requestHeaders',
-      'urlRoot'
+      'urlRoot',
+      'superagent'
     ]))
   },
-	
+
 	request(options = {}) {
 
 		let { method, data, endpoint, onSuccess, onError, fileData } = options;
     let requestData, requestHeaders, doRequest;
+    const request = this.superagent || superagent;
```

Comment:
Yes Vijay, but you are breaking existing code style — variables that can be passed as options are explicitly assigned in destructive assignment above, it is a way of documenting code, to be aware of possible options. superagent option is not assigned that way, that's what I'm complaining about. Thank you for PR btw =)